File: backend/app/schemas/entry.py
"""
文章条目相关 Schema
"""
from datetime import datetime
from typing import Optional, List

from pydantic import BaseModel, Field

from app.models.entry import EntryStatus


class EntryBase(BaseModel):
    """条目基础字段"""
    title: str
    link: str
    author: Optional[str] = None
    published_at: Optional[datetime] = None
    content: Optional[str] = None


class EntryResponse(BaseModel):
    """条目响应"""
    id: int
    rss_source_id: Optional[int]  # 可空，源被删除后为 None
    title: str
    link: str
    author: Optional[str]
    published_at: Optional[datetime]
    content: Optional[str]
    content_type: str
    status: EntryStatus
    is_read: bool
    marked_at: Optional[datetime]
    ai_summary: Optional[str]
    ai_content_type: Optional[str]
    ai_processed_at: Optional[datetime]
    user_notes: Optional[str]
    exported_to_zotero: bool
    fetched_at: datetime
    created_at: datetime

    # 关联信息（即使源被删除，仍保留源名称）
    rss_source_name: Optional[str] = None

    model_config = {"from_attributes": True}


class EntryListResponse(BaseModel):
    """条目列表响应"""
    items: List[EntryResponse]
    total: int
    page: int
    page_size: int
    has_more: bool


class EntryStatusUpdateRequest(BaseModel):
    """更新条目状态请求"""
    status: EntryStatus


class EntryBatchStatusRequest(BaseModel):
    """批量更新条目状态请求"""
    ids: List[int] = Field(..., min_length=1)
    status: EntryStatus


class EntryBatchResponse(BaseModel):
    """批量操作响应"""
    updated_count: int


class EntryNotesUpdateRequest(BaseModel):
    """更新条目笔记请求"""
    notes: Optional[str] = None


class EntryStatsResponse(BaseModel):
    """条目统计响应"""
    total: int
    by_status: dict[str, int]
    today_count: int
    unread_count: int

File: frontend/src/types/entry.ts
// Backend entry status enum values
export type EntryStatus = 'unread' | 'interested' | 'trash' | 'favorite' | 'archived';

// Unified task processing status
export type TaskStatus = 'pending' | 'running' | 'completed' | 'failed' | 'skipped';

// Backward-compatible alias
export type TranslationStatus = TaskStatus;

// Backend entry response type
export interface Entry {
  id: number;
  rss_source_id: number;
  title: string;
  link: string;
  author: string | null;
  published_at: string | null;
  content: string | null;
  content_type: string;
  status: EntryStatus;
  is_read: boolean;
  marked_at: string | null;
  ai_summary: string | null;
  ai_processed_at: string | null;
  task_interpret_status: TaskStatus | null;
  task_translation_status: TaskStatus | null;
  user_notes: string | null;
  exported_to_zotero: boolean;
  fetched_at: string;
  created_at: string;
  rss_source_name: string | null;
  translated_abstract: string | null;
  brief_summary: string | null;
}

export interface EntryListResponse {
  items: Entry[];
  total: number;
  page: number;
  page_size: number;
  has_more: boolean;
}

export interface EntryStatsResponse {
  total: number;
  by_status: Record<string, number>;
  today_count: number;
  unread_count: number;
}

// Frontend article type (for display, mapped from Entry)
export interface Article {
  id: string;
  title: string;
  source: string;
  author: string;
  snippet: string;
  content: string;
  timestamp: string;
  status: 'inbox' | 'saved' | 'discarded';
  tags: string[];
  summary?: string;
  readTime?: string;
  isFavorite?: boolean;
  url?: string;
  // Keep original entry data for API calls
  _entry?: Entry;
}
